src/Extensions/Compatibility/Logger/ULSLogging.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Concurrent;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Omex.Extensions.Compatibility.Logger
+{
+	/// <summary>
+	/// Unified logging, allows registering for logging events sent by
+	/// code implementations and to raise logging events
+	/// </summary>
+	public static class ULSLogging
+	{
+		/// <summary>
+		/// Logs a trace
+		/// </summary>
+		/// <param name="eventId">Unique tag id for the trace</param>
+		/// <param name="category">Logging category</param>
+		/// <param name="level">Logging level</param>
+		/// <param name="message">The message, or message format string, to log</param>
+		/// <param name="parameters">Message format parameters</param>
+		[Obsolete(ObsoleteMessage, IsObsoleteError)]
+		public static void LogTraceTag(EventId eventId, Category category, LogLevel level, string message, params object[] parameters) =>
+			GetLogger(category.Name).Log(level, eventId, message, parameters);
+
+
+		/// <summary>
+		/// Report an exception
+		/// </summary>
+		/// <param name="eventId">Unique tag id for the trace</param>
+		/// <param name="category">Logging category</param>
+		/// <param name="exception">Exception to log</param>
+		/// <param name="message">The message, or message format string, to log</param>
+		/// <param name="parameters">Message format parameters</param>
+		[Obsolete(ObsoleteMessage, IsObsoleteError)]
+		public static void ReportExceptionTag(EventId eventId, Category category, Exception exception, string message, params object[] parameters) =>
+			GetLogger(category.Name).LogError(eventId, exception, message, parameters);
+
+
+		internal static void Initialize(ILoggerFactory loggerFactory)
+		{
+			s_factory = loggerFactory;
+			s_loggersDictionary = new ConcurrentDictionary<string, ILogger>(StringComparer.OrdinalIgnoreCase);
```

Comment:
Could you use `ConcurrentDictionary<Category, ILogger>` here instead and avoid the need to convert to a string on each call. If `Category` is an enum, the lookup should be quite efficient.

Reply:
The category is a custom struct with a string inside here, so I would like to use the string directly to avoid any problems with equality implementation in Category. And also inside of a logger it's just a string